homework/grayscale-image/compression.cpp

```diff
@@ -0,0 +1,68 @@
+#include "compression.hpp"
+
+std::vector<std::pair<uint8_t, uint8_t>> compressGrayscale(const std::array<std::array<uint8_t, width>, height>& bitmap) {
+    std::vector<std::pair<uint8_t, uint8_t>> vec {};
+    vec.reserve(width * height);
+    size_t counter = 1;
+    for (const auto& line : bitmap) {
+        for (auto it = cbegin(line); it < cend(line); ++it) {
+            if (it != (cend(line) - 1) && *it == *(it + 1)) {
+                counter++;
+            } else {
+                vec.push_back(std::make_pair(*it, counter));
+                counter = 1;
+            }
+        }
+    }
+    return vec;
+}
+
+std::array<std::array<uint8_t, width>, height> decompressGrayscale(const std::vector<std::pair<uint8_t, uint8_t>>& bitmap) {
+    std::array<std::array<uint8_t, width>, height> arr;
+    auto counter = 0;
+    auto iter = bitmap.begin();
+    int next = static_cast<int>(iter->second);
+    while (++counter < width * height) {
+        if (counter == next) {
+            iter++;
+            next += iter->second;
+        }
+        arr[counter / width][counter % width] = iter->first;
+    }
```

Comment:
You can also read about array.fill(). It has special overload function for 2D array, which allow to fill more then one row :)